src/Database/DataFeed.php

```diff
@@ -211,9 +217,15 @@ protected function processCollection()
             /*
              * Flush the select, add ID and tag
              */
-            $cleanQuery = $cleanQuery->select(Db::raw($keyName." as id"));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT '".$tag."') as ".$this->tagVar));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT ".$sorting.") as ".$this->sortVar));
+
+            $cast = $this->getDbDialect() === 'pgsql' ? 'text' : 'binary';
```

Comment:
@dimensi You can get the driver name directly from the connection - saving you from having to use your custom `getDbDialect()` method.
```suggestion
            $cast = (Db::connection()->getDriverName() === 'pgsql') ? 'text' : 'binary';
```